docs/source/user-guide/vision/custom_checks.py

```diff
@@ -0,0 +1,209 @@
+result
+
+#%%
+# Our check ran successfully, but we got the print “Nothing found”. This is because we haven’t defined to the check
+# anything to display, so the default behavior is to print “Nothing found”. In order to access the value that we have
+# defined earlier we can use the “value” property on the result.
+
+result.value
+
+#TODO: how does this LINK work?
+
+# %% To see code references for more complex checks (that can receive parameters etc.), check out any of your
+# favorite checks from our API Reference (LINK).
+
+#%%
+# Check Display
+# ========================
+#
+# Most of the times we will want our checks to have a visual display that will quickly summarize the check result. We
+# can pass objects for display to the CheckResult. Objects for display should be of type: html string, dataframe or a
+# function that plots a graph. Let’s define a graph that will be displayed using `Plotly <https://plotly.com/>`_. We
+# will inherit from the original check to shorten the code an update only the compute method.
+#
+# *Good to know: ``display`` can receive a single object to display or a list of objects*
+
+import pandas as pd
+import plotly.express as px
+
+
+class ColorAveragesHistCheck(ColorAveragesCheck):
+
+    def compute(self, context: Context):
+        """Compute the color averages and return them. Also display a histogram comparing train and test."""
+        # Divide by the number of pixels to get the average pixel value per color channel
+        for dataset_kind in DatasetKind:
+            self._color_averages[dataset_kind.value] /= self._pixel_count[dataset_kind.value]
+        # Return the color averages in a dict by channel name
+        return_value = {d_kind: dict(zip(self.channel_names, color_averages))
+                        for d_kind, color_averages in self._color_averages.items()}
+
+        # Display a histogram comparing train and test
+        color_averages_df = pd.DataFrame(return_value).unstack().reset_index()
+        color_averages_df.columns = ['Dataset', 'Channel', 'Pixel Value']
+        fig = px.histogram(color_averages_df, x='Dataset', y='Pixel Value', color='Channel', barmode='group',
+                           histfunc='avg', color_discrete_sequence=['rgb(255,0,0)', 'rgb(0,255,0)', 'rgb(0,0,255)'],
+                           title='Color Averages Histogram')
+        return CheckResult(return_value, display=[fig])
+
+#%%
+# Let check it out:
+
+result = ColorAveragesHistCheck().run(train_ds, test_ds)
+result
+
+# %%
+# Voilà! Now we have a check that prints a graph and has a value. We can add this check
+# to any Suite, and it will run within it.
+
+# %%
+# Defining a Condition
+# ========================
+#
+# Finally, we can add a condition to our check. A condition is a function that receives the result of the check and
+# returns a condition result object. To read more on conditions, check out user guide (LINK). # TODO: how to link to configure a check condition user guide?
```

Comment:
links in rst are in the following:

```
`user guide <../general/customizations/configure_check_conditions>`_

Reply:
need `ref: ` before?

Reply:
Sometimes. Let's try without it and see.